crates/store/re_remote_store_types/src/lib.rs

```diff
@@ -0,0 +1,339 @@
+                    name: "component".to_owned(),
+                }),
+            }),
+            column_selection: Some(ColumnSelection {
+                columns: vec![ColumnSelector {
+                    selector_type: Some(SelectorType::ComponentColumn(ComponentColumnSelector {
+                        entity_path: Some(EntityPath {
+                            path: "/somepath/c".to_owned(),
+                        }),
+                        component: Some(Component {
+                            name: "component".to_owned(),
+                        }),
+                    })),
+                }],
+            }),
+            sparse_fill_strategy: SparseFillStrategy::None.into(),
+        };
+
+        // to chunk store query expression
+        let expected_qe = re_dataframe::external::re_chunk_store::QueryExpression {
+            view_contents: Some(BTreeMap::from([(
+                re_log_types::EntityPath::from("/somepath"),
+                Some(BTreeSet::from([
+                    re_dataframe::external::re_chunk::ComponentName::new("component"),
+                ])),
+            )])),
+            filtered_index: re_log_types::Timeline::new_temporal("log_time"),
+            filtered_index_range: Some(re_dataframe::external::re_chunk_store::IndexRange::new(
+                0, 100,
+            )),
+            filtered_index_values: Some(
+                vec![0, 1, 2]
+                    .into_iter()
+                    .map(re_log_types::TimeInt::new_temporal)
+                    .collect::<BTreeSet<_>>(),
+            ),
+            using_index_values: Some(
+                vec![3, 4, 5]
+                    .into_iter()
+                    .map(re_log_types::TimeInt::new_temporal)
+                    .collect::<BTreeSet<_>>(),
+            ),
+            filtered_point_of_view: Some(
+                re_dataframe::external::re_chunk_store::ComponentColumnSelector {
+                    entity_path: re_log_types::EntityPath::from("/somepath/c"),
+                    component: re_dataframe::external::re_chunk::ComponentName::new("component"),
+                    join_encoding: re_dataframe::external::re_chunk_store::JoinEncoding::default(),
+                },
+            ),
+            sparse_fill_strategy:
+                re_dataframe::external::re_chunk_store::SparseFillStrategy::default(),
+            selection: Some(vec![
+                re_dataframe::external::re_chunk_store::ComponentColumnSelector {
+                    entity_path: re_log_types::EntityPath::from("/somepath/c"),
+                    component: re_dataframe::external::re_chunk::ComponentName::new("component"),
+                    join_encoding: re_dataframe::external::re_chunk_store::JoinEncoding::default(),
+                }
+                .into(),
+            ]),
+        };
+
+        let query_expression: re_dataframe::external::re_chunk_store::QueryExpression =
+            query.try_into().unwrap();
```

Comment:
A full roundtrip would be nice:
```
let query_grpc_before = /* ... */;
let query_native = query_grpc_before.try_into();
let query_grpc_after =  query_native.into();
assert_eq!(query_grpc_before, query_grpc_after);
```
at which point you can probably remove the handwritten expectations

Reply:
that makes sense, will add the other way conversion and update the unit test